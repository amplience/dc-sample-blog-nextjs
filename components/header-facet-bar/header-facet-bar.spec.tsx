import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import HeaderFacetBar from './header-facet-bar';

jest.mock('react-instantsearch-dom', () => ({
  ...jest.requireActual('react-instantsearch-dom')
}));
describe('HeaderFaceBar component', () => {
  it('should render author and tags facet menus with a clear button', async () => {
    const component = ShallowRenderer.createRenderer();
    const props = { tags: 'tags', authors: 'authors' };
    component.render(<HeaderFacetBar {...props} />);
    expect(component.getRenderOutput()).toMatchSnapshot();
  });
});
