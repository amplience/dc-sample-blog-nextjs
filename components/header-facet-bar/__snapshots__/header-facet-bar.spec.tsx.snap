// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFaceBar component should render author and tags facet menus with a clear button 1`] = `
<React.Fragment>
  <div
    className="jsx-39763164"
    id="FacetBar"
  >
    <h3
      className="jsx-39763164"
    >
      Show blogs by
    </h3>
    <div
      className="jsx-39763164 filter-container"
    >
      <img
        alt="Authors"
        className="jsx-39763164"
        src="/static/images/ic-person.svg"
      />
      <ConnectorWrapper
        attribute="authors"
        limit={25}
      />
    </div>
    <div
      className="jsx-39763164 filter-container"
    >
      <img
        alt="Tags"
        className="jsx-39763164"
        src="/static/images/ic-tag.svg"
      />
      <ConnectorWrapper
        attribute="tags"
        limit={25}
      />
    </div>
    <ConnectorWrapper
      clearsQuery={true}
    />
  </div>
  <JSXStyle
    dynamic={
      Array [
        "736px",
      ]
    }
    id="3767950373"
  >
    #FacetBar.__jsx-style-dynamic-selector{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;text-align:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-top:20px;}h3.__jsx-style-dynamic-selector{font-size:16px;color:#999;font-weight:400;padding-right:20px;white-space:nowrap;}img.__jsx-style-dynamic-selector{height:40px;width:40px;padding-right:10px;fill:#a7aaaf;}.filter-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media (max-width:736px){#FacetBar.__jsx-style-dynamic-selector{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}h3.__jsx-style-dynamic-selector{margin-bottom:2px;padding-right:0;}.filter-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}}
  </JSXStyle>
</React.Fragment>
`;
